### DISCUSSION TITLE
How to build rspamd as Debian package

### DISCUSSION THREAD

**Discussion by counterpoint** [rspamd/rspamd#4029] [Q&A]
Is any information available on how to build rspamd as a Debian package please? The documentation suggests:
`$ mkdir rspamd.build
$ cd rspamd.build
$ ./dist.sh
$ tar xvf rspamd-<rspamd_version>.tar.xz
$ cd rspamd-<rspamd_version>
$ debuild`
Presumably one is intended to cd into the rspamd directory created by git first of all. But the instructions suggest creating a further directory, but as it is empty, dist.sh isn't found there. Trying to build in the basic rspamd directory fails because a file parameter is required. Inventing a name as parameter allows the process to run, and it creates a file of the given name. But dpkg says it isn't a Debian package. I'm not expert in building things, and don't know where to go from here. Any suggestions please?

**Comment by stevesobol** [comment]
Hello,

I'm just starting to use rspamd, and I installed it from .DEB on a Debian derivative (Ubuntu 18.04, to be specific). It's available in Ubuntu's package repositories.

It's available in Debian's repos too (https://packages.debian.org/search?keywords=rspamd).

Two thoughts:
* you don't _need to_ build your own package if you don't want to. Debian and Ubuntu are the two most popular Debian-esque Linux distros, and both have rspamd packages already. However,
* if you still want to do it yourself, grab the DSC (source package) and take a look at how the package maintainers do it.

I use Ubuntu and Debian almost exclusively, both for my own projects and on Linux servers I maintain for clients, and I've created DEBs before, and I can offer some insight if you need it, but those discussions would probably be off-topic here. If you need help, let me know here, and we'll take the conversation "offline."
